### PR TITLE
Define infix operators on Globals

### PR DESCRIPTION
**Pull request by dham**
This fixes issues where operations on Firedrake Functions in the R space assume that the usual operators are defined for Globals.

### CODE REVIEW

File: pyop2/base.py [hunk 1]
```diff
@@ -2737,6 +2737,84 @@ def halo_exchange_end(self):
...
+
+    def __imul__(self, other):
+        """Pointwise multiplication or scaling of fields."""
+        return self._iop(other, operator.imul)
+
+    def __itruediv__(self, other):
+        """Pointwise division or scaling of fields."""
+        return self._iop(other, operator.itruediv)
+
+    __idiv__ = __itruediv__  # Python 2 compatibility
```
Comment (miklos1) [inline pyop2/base.py]: Are we still supporting Python 2 in PyOP2?

Comment (dham) [inline pyop2/base.py]: Well we haven't disabled it elsewhere, so not putting it in at this point would create an undocumented breakage. However I agree, we should drop Python 2.7 support (but do it in one fell swoop).

Comment (miklos1) [inline pyop2/base.py]: See #530.

Comment (dham) [inline pyop2/base.py]: Happy now?

File: test/unit/test_globals.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
+# This file is part of PyOP2
+#
+# PyOP2 is Copyright (c) 2012, Imperial College London and
```
Comment (miklos1) [inline test/unit/test_globals.py]: Wrong date.

File: pyop2/base.py (hunk 1 above)
Comment (miklos1) [inline pyop2/base.py]: Yes.

**Pull request merged**